plutus-core/plutus-core/src/PlutusCore/Core/Type.hs

```diff
@@ -94,6 +98,38 @@ type family UniOf a :: GHC.Type -> GHC.Type
 
 type instance UniOf (Term tyname name uni fun ann) = uni
 
+-- | A "type variable declaration", i.e. a name and a kind for a type variable.
+data TyVarDecl tyname ann = TyVarDecl
+    { tyVarDeclAnn  :: ann
+    , tyVarDeclName :: tyname
+    , tyVarDeclKind :: Kind ann
+    } deriving (Functor, Show, Generic)
```

Comment:
Had to move these to `PlutusCore/Core/Type.hs` in order to break dependency loops.